javaslang/src/main/java/javaslang/collection/Arrays.java

```diff
@@ -35,6 +39,19 @@
         return array;
     }
 
+    @SuppressWarnings("unchecked")
+    static <T> Object[] asArray(Iterable<? extends T> iterable) {
+        if (iterable instanceof Collection<?>) {
+            final Collection<? extends T> collection = (Collection<? extends T>) iterable;
+            return collection.toArray();
+        } else if (iterable instanceof Traversable) {
+            final Traversable<T> traversable = asTraversableAgain(iterable);
```

Comment:
Please point me to (one of) the tests. Iterator should be the only TraversableOnce (in our code-base).